lang/candidate.go

```diff
@@ -42,6 +42,10 @@ type Candidate struct {
 	// TriggerSuggest allows server to instruct the client whether
 	// to reopen candidate suggestion popup after insertion
 	TriggerSuggest bool
+
+	// ResolveHook allows the async resolution of additional information
```

Comment:
```suggestion
	// ResolveHook allows resolution of additional information
```
I understand the intentions here, but I think it could be potentially misunderstood for protocol-level synchronicity in the context of this library, which is in fact a concern of the consumer (LS) and we don't use any sync primitives in `hcl-lang` really.